SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialPlayerSpawner.cpp

```diff
@@ -156,38 +153,206 @@ void USpatialPlayerSpawner::ReceivePlayerSpawnResponse(const Worker_CommandRespo
 	}
 }
 
-void USpatialPlayerSpawner::ObtainPlayerParams(FURL& LoginURL, FUniqueNetIdRepl& OutUniqueId, FName& OutOnlinePlatformName)
+void USpatialPlayerSpawner::ReceivePlayerSpawnRequestOnServer(const Worker_CommandRequestOp& Op)
 {
-	const FWorldContext* const WorldContext = GEngine->GetWorldContextFromWorld(NetDriver->GetWorld());
-	check(WorldContext->OwningGameInstance);
+	UE_LOG(LogSpatialPlayerSpawner, Log, TEXT("Received PlayerSpawn request on server"));
 
-	// This code is adapted from PendingNetGame.cpp:242
-	if (ULocalPlayer* LocalPlayer = WorldContext->OwningGameInstance->GetFirstGamePlayer())
+	FString WorkerID(UTF8_TO_TCHAR(Op.caller_worker_id));
+
+	// Accept the player if we have not already accepted a player from this worker.
+	bool bAlreadyHasPlayer;
+	WorkersWithPlayersSpawned.Emplace(WorkerID, &bAlreadyHasPlayer);
+	if (bAlreadyHasPlayer)
 	{
-		// Send the player nickname if available
-		FString OverrideName = LocalPlayer->GetNickname();
-		if (OverrideName.Len() > 0)
+		UE_LOG(LogSpatialPlayerSpawner, Verbose, TEXT("Ignoring duplicate PlayerSpawn request. Client worker ID: %s", *ClientWorkerID));
```

Comment:
Several of the UE_LOGs in this file have an issue where the TEXT macro isn't closed before the args:
UE_LOG(LogSpatialPlayerSpawner, Verbose, TEXT("text %s", *ClientWorkerID)); // broken
UE_LOG(LogSpatialPlayerSpawner, Verbose, TEXT("text %s"), *ClientWorkerID); // fixed

Reply:
also needs to be WorkerID to match the variable name on line 160